checks.d/consul.py

```diff
@@ -323,3 +361,69 @@ def check(self, instance):
                         status_value,
                         tags=main_tags+node_tags
                     )
+
+        if perform_network_latency_checks:
+            self.check_network_latency(instance, agent_dc, main_tags)
+
+    def _get_coord_datacenters(self, instance):
+        return self.consul_request(instance, '/v1/coordinate/datacenters')
+
+    def _get_coord_nodes(self, instance):
+        return self.consul_request(instance, 'v1/coordinate/nodes')
+
+    def check_network_latency(self, instance, agent_dc, main_tags):
+
+        datacenters = self._get_coord_datacenters(instance)
+        for datacenter in datacenters:
+            name = datacenter['Datacenter']
+            if name != agent_dc:
+                # Not us, skip
+                continue
+            # Inter-datacenter
+            for other in datacenters:
+                other_name = other['Datacenter']
+                if name == other_name:
+                    # Ignore ourself
+                    continue
+                latencies = []
+                for node_a in datacenter['Coordinates']:
+                    for node_b in other['Coordinates']:
+                        latencies.append(distance(node_a, node_b))
+                latencies.sort()
+                tags = main_tags + ['source_datacenter:{}'.format(name),
+                                    'dest_datacenter:{}'.format(other_name)]
+                n = len(latencies)
+                half_n = int(floor(n / 2))
+                if n % 2:
+                    median = latencies[half_n]
+                else:
+                    median = (latencies[half_n - 1] + latencies[half_n]) / 2
+                self.gauge('consul.net.dc-latency.min', latencies[0], hostname='', tags=tags)
+                self.gauge('consul.net.dc-latency.median', median, hostname='', tags=tags)
+                self.gauge('consul.net.dc-latency.max', latencies[-1], hostname='', tags=tags)
```

Comment:
I'm going to have to check how this `hostname=''` parameter might play with our backend. I get why you're overriding (because they're not host metrics), but `key:value` tags with an empty value might not be desireable.